drivers/devicetree/fdt.c

```diff
@@ -84,3 +85,159 @@ FAR const char *fdt_get(void)
 {
   return g_fdt_base;
 }
+
+/****************************************************************************
+ * Name: fdt_get_irq
+ *
+ * Description:
+ *   Get the interrupt number of the node
+ *
+ ****************************************************************************/
+
+int fdt_get_irq(FAR const void *fdt, int offset, int irqbase)
```

Comment:
Does the FDT address need to be supplied externally? Since once the FDT is registered, its address is available in this file through g_fdt_base.

Reply:
I think user maybe use both new APIs and original APIs,
so better keep the void *fdt handle same.

Maybe have a little difficult to read like this:

```
      offset = fdt_node_offset_by_compatible(fdt, offset, "virtio,mmio");  // must use fdt
      if (offset == -FDT_ERR_NOTFOUND)
        {
          break;
        }

      addr = fdt_get_reg_base(offset);  // don't use fdt
      irqnum = fdt_get_irq(offset, QEMU_SPI_IRQ_BASE);
      if (addr > 0 && irqnum >= 0)
        {
          virtio_register_mmio_device((void *)addr, irqnum);
        }

```